test/common/upstream/health_check_fuzz.cc

```diff
@@ -0,0 +1,202 @@
+  if (!test_sessions_[0]->interval_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Interval timer is disabled. Skipping trigger interval timer.");
+    return;
+  }
+  expectStreamCreate(0);
+  ENVOY_LOG_MISC(trace, "Triggered interval timer");
+  test_sessions_[0]->interval_timer_->invokeCallback();
+}
+
+void HealthCheckFuzz::triggerTimeoutTimer(bool last_action) {
+  // Timeout timer needs to be explicitly enabled, usually by a call to onIntervalBase().
+  if (!test_sessions_[0]->timeout_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Timeout timer is disabled. Skipping trigger timeout timer.");
+    return;
+  }
+  ENVOY_LOG_MISC(trace, "Triggered timeout timer");
+  test_sessions_[0]->timeout_timer_->invokeCallback(); // This closes the client, turns off timeout
+                                                       // and enables interval
+  if (!last_action) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream from network timeout");
+    expectClientCreate(0);
+    expectStreamCreate(0);
+    test_sessions_[0]->interval_timer_->invokeCallback();
+  }
+}
+
+void HealthCheckFuzz::raiseEvent(const test::common::upstream::RaiseEvent& event,
+                                 bool last_action) {
+  Network::ConnectionEvent eventType;
+  switch (event) {
+  case test::common::upstream::RaiseEvent::CONNECTED: {
+    eventType = Network::ConnectionEvent::Connected;
+    break;
+  }
+  case test::common::upstream::RaiseEvent::REMOTE_CLOSE: {
+    eventType = Network::ConnectionEvent::RemoteClose;
+    break;
+  }
+  case test::common::upstream::RaiseEvent::LOCAL_CLOSE: {
+    eventType = Network::ConnectionEvent::LocalClose;
+    break;
+  }
+  default: // shouldn't hit
+    eventType = Network::ConnectionEvent::Connected;
+    break;
+  }
+
+  switch (type_) {
+  case HealthCheckFuzz::Type::HTTP: {
+    test_sessions_[0]->client_connection_->raiseEvent(eventType);
+    if (!last_action && eventType != Network::ConnectionEvent::Connected) {
+      ENVOY_LOG_MISC(trace, "Creating client and stream from close event");
+      expectClientCreate(0);
+      expectStreamCreate(0);
+      test_sessions_[0]->interval_timer_->invokeCallback();
+    }
+    break;
+  }
+  default:
+    break;
+  }
+}
+
+void HealthCheckFuzz::replay(const test::common::upstream::HealthCheckTestCase& input) {
+  for (int i = 0; i < input.actions().size(); ++i) {
+    const auto& event = input.actions(i);
+    const bool last_action = i == input.actions().size() - 1;
+    ENVOY_LOG_MISC(trace, "Action: {}", event.DebugString());
+    switch (event.action_selector_case()) { // TODO: Once added implementations for tcp and gRPC,
+                                            // move this to a separate method, handleHttp
+    case test::common::upstream::Action::kRespond: {
+      switch (type_) {
+      case HealthCheckFuzz::Type::HTTP: {
+        // TODO: Hardcoded check on status Required because can't find documentation about required
```

Comment:
A string is non-empty when it has length one :)

Reply:
Critical thinking!